packages/react-tinacms-inline/src/blocks/add-block-menu.tsx

```diff
@@ -19,18 +19,45 @@ limitations under the License.
 import * as React from 'react'
 import styled, { css } from 'styled-components'
 import { BlockTemplate } from 'tinacms'
-import { Button } from '@tinacms/styles'
+import { IconButton } from '@tinacms/styles'
 import { AddIcon } from '@tinacms/icons'
+import { Dismissible } from 'react-dismissible'
 
 interface AddBlockMenuProps {
   addBlock(data: any): void
   templates: BlockTemplate[]
+  position?: 'top' | 'bottom'
+  index?: number
+  offset?: number
 }
 
-export function AddBlockMenu({ templates, addBlock }: AddBlockMenuProps) {
+export function AddBlockMenu({
+  templates,
+  addBlock,
+  position,
+  index,
+  offset,
+}: AddBlockMenuProps) {
   const [isOpen, setIsOpen] = React.useState(false)
+  const addBlockButtonRef = React.useRef<HTMLButtonElement>(null)
+  const [openTop, setOpenTop] = React.useState(false)
 
   const handleOpenBlockMenu = () => {
+    const addBlockButtonElem = addBlockButtonRef.current
+
+    if (addBlockButtonElem !== null) {
+      const menuBounding = addBlockButtonElem.getBoundingClientRect()
+      const halfWindowHeight =
+        (window.innerHeight || document.documentElement.clientHeight) / 2
```

Comment:
I'm guessing this is related to the error. We should check whether `window` or `document` is available before running this. Maybe put it in a componentDidMount-style useEffect?

Reply:
Looks like it's from the dismissible component itself. You can provide it with a document if you're inside an iFrame (I believe that's what it's for?) but in this case it didn't seem necessary. Strange.

Reply:
Yeah I can't find another solution, I just removed dismissible for now. It's functionally almost identical.